extensions/canvasMedical/actions/createQuestionnaireResponses.ts

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ZodError } from 'zod'
+import {
+  FieldType,
+  type Action,
+  type DataPointDefinition,
+  type Fields,
+} from '@awell-health/extensions-core'
+import { Category } from '@awell-health/extensions-core'
+import { type settings } from '../settings'
+import { makeAPIClient } from '../client'
+import { fromZodError } from 'zod-validation-error'
+import { AxiosError } from 'axios'
+import { creatingQuestionnaireResponsesSchema } from '../validation/dto/questionnaireResponses.zod'
+import type schemas from '../schemas'
+
+const fields = {
+  questionnaireId: {
+    id: 'questionnaireId',
+    label: 'QuestionnaireId',
+    description:
+      'Reference to the Canvas Questionnaire using the questionnaire id',
+    type: FieldType.STRING,
+    required: true,
+  },
+  subjectId: {
+    id: 'subjectId',
+    label: 'SubjectId',
+    description: 'Reference to the Canvas Patient using the patient id',
+    type: FieldType.STRING,
+    required: true,
+  },
+  authored: {
+    id: 'authored',
+    label: 'Authored',
+    description:
+      'Timestamp the Questionnaire response was filled out (If omitted the current timestamp at data ingestion will be used)',
+    type: FieldType.STRING,
+    required: false,
+  },
+  authorId: {
+    id: 'authorId',
+    label: 'AuthorId',
+    description:
+      'Reference to the patient or practitioner filling out the questionnaire. If omitted it will default to Canvas Bot',
+    type: FieldType.STRING,
+    required: false,
+  },
+  item: {
+    id: 'item',
+    label: 'Item',
+    description: 'List of answers to questions in the questionnaire',
+    type: FieldType.JSON,
+    required: true,
+  },
+} satisfies Fields<typeof schemas>
+
+const dataPoints = {
+  taskId: {
+    key: 'QuestionnaireResponseId',
+    valueType: 'string',
+  },
+} satisfies Record<string, DataPointDefinition>
+
+const createPayload = (
+  questionnaireId?: string,
+  subjectId?: string,
+  authored?: string,
+  authorId?: string,
+  item?: any[]
+): any => {
+  const payload = {
+    resourceType: 'QuestionnaireResponse' as 'QuestionnaireResponse',
+    questionnaire:
+      questionnaireId !== undefined
+        ? `Questionnaire/${questionnaireId}`
+        : undefined,
+    subject:
+      subjectId !== undefined
+        ? {
+            reference: `Patient/${subjectId}`,
+          }
+        : undefined,
+    item,
+    authored,
+    author:
+      authorId !== undefined
+        ? {
+            reference: `Practitioner/${authorId}`,
+          }
+        : undefined,
```

Comment:
How do we distinguish if the id provided for author is a patient or a practitioner? Here you are assuming it is a practitioner but from the field above it looks like it could be either id?

Either the user should specify somehow which type of id they are providing here, or we should just let the update be done by Canvas bot always